Scaling will throw PipelineJobPrepareFailedException, when start/stop scaling interval is very short

## Bug Report

The command is executed as follows
```
mysql> CREATE TABLE t_order (order_id bigint NOT NULL, user_id int DEFAULT NULL, status varchar(50) DEFAULT NULL, t_numeric numeric(10,2) DEFAULT NULL, PRIMARY KEY (order_id));
Query OK, 0 rows affected (0.60 sec)

mysql> ALTER SHARDING TABLE RULE t_order(RESOURCES(ds_2),SHARDING_COLUMN=order_id,TYPE(NAME=hash_mod,PROPERTIES('sharding-count'=6)),KEY_GENERATE_STRATEGY(COLUMN=order_id,TYPE(NAME=snowflake)));
Query OK, 0 rows affected (2.10 sec)

mysql> check scaling 0130317c30317c3054317c7368617264696e675f6462;
+------------+----------------------+----------------------+-----------------------+-------------------------+
| table_name | source_records_count | target_records_count | records_count_matched | records_content_matched |
+------------+----------------------+----------------------+-----------------------+-------------------------+
| t_order    | 367                  | 0                    | false                 | false                   |
+------------+----------------------+----------------------+-----------------------+-------------------------+
1 row in set (1.06 sec)

mysql> check scaling 0130317c30317c3054317c7368617264696e675f6462;
+------------+----------------------+----------------------+-----------------------+-------------------------+
| table_name | source_records_count | target_records_count | records_count_matched | records_content_matched |
+------------+----------------------+----------------------+-----------------------+-------------------------+
| t_order    | 554                  | 69                   | false                 | false                   |
+------------+----------------------+----------------------+-----------------------+-------------------------+
1 row in set (0.54 sec)

mysql> STOP SCALING 0130317c30317c3054317c7368617264696e675f6462;
Query OK, 0 rows affected (0.05 sec)

mysql> start scaling 0130317c30317c3054317c7368617264696e675f6462;
Query OK, 0 rows affected (0.02 sec)

mysql> STOP SCALING 0130317c30317c3054317c7368617264696e675f6462;
Query OK, 0 rows affected (0.02 sec)

mysql> show scaling list;
+----------------------------------------------+---------+----------------------+--------+---------------------+---------------------+
| id                                           | tables  | sharding_total_count | active | create_time         | stop_time           |
+----------------------------------------------+---------+----------------------+--------+---------------------+---------------------+
| 0130317c30317c3054317c7368617264696e675f6462 | t_order | 2                    | false  | 2022-06-20 13:44:23 | 2022-06-20 13:44:46 |
+----------------------------------------------+---------+----------------------+--------+---------------------+---------------------+
1 row in set (0.06 sec)

mysql> show scaling status 0130317c30317c3054317c7368617264696e675f6462;
+------+-------------+-------------------+--------+-------------------------------+--------------------------+
| item | data_source | status            | active | inventory_finished_percentage | incremental_idle_seconds |
+------+-------------+-------------------+--------+-------------------------------+--------------------------+
| 0    | ds_0        | RUNNING           | false  | 0                             | 0                        |
| 1    |             | PREPARING_FAILURE | false  | 0                             | 0                        |
+------+-------------+-------------------+--------+-------------------------------+--------------------------+
2 rows in set (0.04 sec)
```

When execute start/stop scaling

### Which version of ShardingSphere did you use?

master

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy

### Expected behavior

no exception will be thrown.

### Actual behavior

```
[ERROR] 2022-06-20 13:44:50.903 [Thread-6] o.a.s.d.p.s.r.RuleAlteredJobScheduler - job prepare failed, 0130317c30317c3054317c7368617264696e675f6462-1
org.apache.shardingsphere.data.pipeline.core.exception.PipelineJobPrepareFailedException: Job stopping, jobId=0130317c30317c3054317c7368617264696e675f6462
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobPreparer.prepare(RuleAlteredJobPreparer.java:84)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobScheduler.run(RuleAlteredJobScheduler.java:74)
	at java.lang.Thread.run(Thread.java:750)
```

### Reason analyze (If you can)

The stop event is not handled correctly and affects the next start

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/exception/PipelineIgnoredException.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/exception/PipelineIgnoredException.java
@@ -1,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.data.pipeline.core.exception;
+
+/**
+ * Pipeline job ignored exception.
+ */
+public final class PipelineIgnoredException extends RuntimeException {
+    
+    private static final long serialVersionUID = -606723977923290937L;
+    
+    public PipelineIgnoredException(final String message) {
+        super(message);
+    }
+    
+    public PipelineIgnoredException(final String message, final Throwable cause) {
+        super(message, cause);
+    }
+}
+

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPreparer.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPreparer.java
@@ -30,6 +30,7 @@ import org.apache.shardingsphere.data.pipeline.api.job.JobStatus;
 import org.apache.shardingsphere.data.pipeline.api.job.progress.JobProgress;
 import org.apache.shardingsphere.data.pipeline.core.context.PipelineContext;
 import org.apache.shardingsphere.data.pipeline.core.datasource.PipelineDataSourceManager;
+import org.apache.shardingsphere.data.pipeline.core.exception.PipelineIgnoredException;
 import org.apache.shardingsphere.data.pipeline.core.exception.PipelineJobPrepareFailedException;
 import org.apache.shardingsphere.data.pipeline.core.execute.ExecuteEngine;
 import org.apache.shardingsphere.data.pipeline.core.ingest.position.PositionInitializerFactory;
@@ -77,11 +78,11 @@ public final class RuleAlteredJobPreparer {
     public void prepare(final RuleAlteredJobContext jobContext) {
         checkSourceDataSource(jobContext);
         if (jobContext.isStopping()) {
-            throw new PipelineJobPrepareFailedException("Job stopping, jobId=" + jobContext.getJobId());
+            throw new PipelineIgnoredException("Job stopping, jobId=" + jobContext.getJobId());
         }
         prepareAndCheckTargetWithLock(jobContext);
         if (jobContext.isStopping()) {
-            throw new PipelineJobPrepareFailedException("Job stopping, jobId=" + jobContext.getJobId());
+            throw new PipelineIgnoredException("Job stopping, jobId=" + jobContext.getJobId());
         }
         // TODO check metadata
         try {

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobScheduler.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobScheduler.java
@@ -24,6 +24,7 @@ import org.apache.shardingsphere.data.pipeline.api.ingest.position.FinishedPosit
 import org.apache.shardingsphere.data.pipeline.api.job.JobStatus;
 import org.apache.shardingsphere.data.pipeline.core.api.GovernanceRepositoryAPI;
 import org.apache.shardingsphere.data.pipeline.core.api.PipelineAPIFactory;
+import org.apache.shardingsphere.data.pipeline.core.exception.PipelineIgnoredException;
 import org.apache.shardingsphere.data.pipeline.core.execute.ExecuteCallback;
 import org.apache.shardingsphere.data.pipeline.core.task.IncrementalTask;
 import org.apache.shardingsphere.data.pipeline.core.task.InventoryTask;
@@ -72,6 +73,10 @@ public final class RuleAlteredJobScheduler implements Runnable {
         GovernanceRepositoryAPI governanceRepositoryAPI = PipelineAPIFactory.getGovernanceRepositoryAPI();
         try {
             jobContext.getJobPreparer().prepare(jobContext);
+        } catch (final PipelineIgnoredException ex) {
+            log.info("pipeline ignore exception: {}", ex.getMessage());
+            RuleAlteredJobCenter.stop(jobId);
+            ShardingSphereEventBus.getInstance().post(new ScalingReleaseDatabaseLevelLockEvent(jobContext.getJobConfig().getDatabaseName()));
             // CHECKSTYLE:OFF
         } catch (final RuntimeException ex) {
             // CHECKSTYLE:ON

